local_server/common/rest.py

```diff
@@ -373,10 +374,39 @@ def genesets_put(request, data_adaptor):
         if genesets is None:
             abort(HTTPStatus.BAD_REQUEST)
 
-        (gs, _) = data_adaptor.check_new_gene_sets((genesets, tid))
-        annotations.write_gene_sets(gs, tid, data_adaptor)
+        annotations.write_gene_sets(genesets, tid, data_adaptor)
         return make_response(jsonify({"status": "OK"}), HTTPStatus.OK)
     except (ValueError, DisabledFeatureError, KeyError) as e:
         return abort_and_log(HTTPStatus.BAD_REQUEST, str(e), include_exc_info=True)
     except (ObsoleteRequest, TypeError) as e:
         return abort(HTTPStatus.NOT_FOUND, description=str(e))
+
+
+def geneset_summary_get(request, data_adaptor):
+    preferred_mimetype = request.accept_mimetypes.best_match(["application/octet-stream"])
+    if preferred_mimetype != "application/octet-stream":
+        return abort(HTTPStatus.NOT_ACCEPTABLE)
+
+    geneset_name = request.args.get("geneset_name", default=None)
+    summary_method = request.args.get("method", default="mean")
+    request_tid = request.args.get("tid", default=None)
+
+    try:
+        annotations = data_adaptor.dataset_config.user_annotations
+        (genesets, tid) = annotations.read_gene_sets(data_adaptor)
+
+        if request_tid is not None and int(request_tid) != tid:
```

Comment:
Quick question about the case where the tid is None -- is this actually a valid scenario where a gene set summary can be fetched? The initial gene sets need to be read before a summary can be fetched (I think?) so the TID is guaranteed to be 0 based on a line in the critciial section of your `read_gene_sets` function where `tid = self.last_geneset_tid`.

Reply:
Ahh I see. I did not grok that if the `request_tid` is None that the client does not want synchronization. Thanks for the clarification.